packages/devtools_app/lib/src/screens/vm_developer/object_inspector/vm_instance_display.dart

```diff
@@ -0,0 +1,244 @@
+// Copyright 2023 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:vm_service/vm_service.dart';
+
+import '../../../shared/common_widgets.dart';
+import '../../../shared/console/widgets/expandable_variable.dart';
+import '../../../shared/globals.dart';
+import '../../../shared/object_tree.dart';
+import '../../../shared/primitives/utils.dart';
+import '../../../shared/split.dart';
+import '../../../shared/theme.dart';
+import '../vm_developer_common_widgets.dart';
+import 'object_inspector_view_controller.dart';
+import 'vm_object_model.dart';
+
+class VmInstanceDisplay extends StatefulWidget {
+  const VmInstanceDisplay({
+    required this.controller,
+    required this.instance,
+  });
+
+  final ObjectInspectorViewController controller;
+  final InstanceObject instance;
+
+  @override
+  State<StatefulWidget> createState() => _VmInstanceDisplayState();
+}
+
+class _VmInstanceDisplayState extends State<VmInstanceDisplay> {
+  late Future<void> _initialized;
+  late DartObjectNode _root;
+
+  @override
+  void initState() {
+    super.initState();
+    _populate();
+  }
+
+  @override
+  void didUpdateWidget(VmInstanceDisplay oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (widget.instance == oldWidget.instance) {
+      return;
+    }
+    _populate();
+  }
+
+  void _populate() async {
+    final isolateRef = serviceManager.isolateManager.selectedIsolate.value;
+    _root = DartObjectNode.fromValue(
+      name: 'value',
+      value: widget.instance.obj,
+      isolateRef: isolateRef,
+      artificialName: true,
+    );
+
+    _initialized = buildVariablesTree(_root)
+        .then(
+          (_) => _root.expand(),
+        )
+        .then(
+          (_) => Future.wait([
+            for (final child in _root.children) buildVariablesTree(child),
```

Comment:
do we want to do this lazily? or is it lazy to 2 levels in its current state?

Reply:
It's lazy to 2 levels (the root object node and its fields).